Bug fix logging

binya2 opened the pull request.

binya2 committed:
fix black ai player

binya2 committed:
fix chess bug

binya2 committed:
fix Knight bug

binya2 committed:
Adding configure log files

binya2 committed:
Print to log file which color started the game && new parameters

binya2 committed:
Added printing to log files who won, how many knight moves were made and how many checkers there were in the game

binya2 committed:
Display the board on the screen during each operation

binya2 committed:
bug fix

binya2 committed:
Checking the knight is not blocked in his steps in the game

binya2 committed:
Test that the knight has no vessel that he can eat, there is nothing to eat

binya2 committed:
A test that checks how many parts can eat around

binya2 committed:
A test that verifies the locations the tool can go

binya2 committed:
A test to get empty positions from the opponent's tools

binya2 committed:
Test to get empty locations from self tools

binya2 committed:
Integration test between locations with different status - full and empty

binya2 committed:
Integration testing for AI calculations

binya2 committed:
General test for moving a tool

talbarfm commented on chess_engine.py:
```diff
@@ -118,6 +118,16 @@ def is_valid_piece(self, row, col):
+    def get_black_king_location(self):
+        return self._black_king_location
+
+    def is_valid_torn(self, row, col):
```
Spelling mistake, 'torn' should be 'turn'

binya2 replied:
I fixed it

binya2 replied:
I fixed it

talbarfm commented on chess_gui.py:
```diff
@@ -119,12 +137,45 @@ def main():
+
     if human_player is 'b':
+        board = "\n---board---\n"
+        for i in range(8):
+            for j in range(8):
+                if game_state.is_valid_piece(i, j):
+                    board += game_state.get_piece(i, j).get_name()
+                    board += " "
+                else:
+                    board += Player.EMPTY
+                    board += " "
+            board += '\n'
```
Try to avoid i and j  if better names are avalible

binya2 replied:
I fixed it

talbarfm commented on chess_gui.py:
```diff
@@ -119,12 +137,45 @@ def main():
         game_state.move_piece(ai_move[0], ai_move[1], True)
 
+    board = "\n---board---\n"
+    for i in range(8):
+        for j in range(8):
+            if game_state.is_valid_piece(i, j):
+                board += game_state.get_piece(i, j).get_name()
+                board += " "
+            else:
+                board += Player.EMPTY
+                board += " "
```
Try to avoid i and j if better names are avalible

binya2 replied:
I fixed it

talbarfm reviewed (changes requested):
ץ

binya2 committed:
Update chess_engine.py

Correcting a spelling mistake

binya2 committed:
Update chess_gui.py

Changing user names in the loop to more relevant names

binya2 commented:
I have corrected all comments

talbarfm commented on chess_gui.py:
```diff
@@ -174,23 +243,44 @@ def main():
                     print(len(game_state.move_log))
-
+        if valid:
+            board = "\n---board---\n"
+           for row in range(8):
+            for col in range(8):
+                if game_state.is_valid_piece(row, col):
+                    board += game_state.get_piece(row, col).get_name()
+                        board += " "
+                    else:
+                        board += Player.EMPTY
+                        board += " "
+                board += '\n'
```
You use this for loop a few times, instead of copying it consider separating it to a different function

binya2 replied:
While writing the code I already tried because the double code bothered me and for some reason it didn't work - I'll try again

binya2 replied:
I made the changes in the repository itself. If there is a problem with the exercise, let me know and upload the code I have on my computer.

talbarfm reviewed (changes requested):
.

talbarfm reviewed (changes requested):
.

binya2 committed:
Update and rename chess_gui.py to chess_gui.pyf

I made the changes in the repository itself. If there is a problem with the exercise, let me know and upload the code I have on my computer.

binya2 commented:
I made the changes in the repository itself. If there is a problem with the exercise, let me know and upload the code I have on my computer.

binya2 committed:
Update and rename chess_gui.pyf to chess_gui.py

I accidentally changed the file name
